oauth2_provider/oauth2_validators.py

```diff
@@ -759,7 +761,17 @@ def get_id_token_dictionary(self, token, token_handler, request):
         return claims, expiration_time
 
     def get_oidc_issuer_endpoint(self, request):
-        return oauth2_settings.oidc_issuer(request)
+        """TODO sc-77176 : Move to greenspace django app"""
+        if oauth2_settings.OIDC_ISS_ENDPOINT:
+            return oauth2_settings.OIDC_ISS_ENDPOINT
+
+        # generate it based on known URL
+        django_request = HttpRequest()
+        django_request.META = request.headers
+
+        abs_url = django_request.build_absolute_uri(reverse("oauth2_provider:oidc-connect-discovery-info"))
+        base_url = abs_url[:-len("/.well-known/openid-configuration/")]
+        return base_url
```

Comment:
Oh nvm we probably don't need this as it happens underneath now... I ran the test locally and it wasn't passing through 🤔.

```
    def oidc_issuer(self, request):
        """
        Helper function to get the OIDC issuer URL, either from the settings
        or constructing it from the passed request.

        If only an oauthlib request is available, a dummy django request is
        built from that and used to generate the URL.
        """
        if self.OIDC_ISS_ENDPOINT:
            return self.OIDC_ISS_ENDPOINT
        if isinstance(request, HttpRequest):
            django_request = request
        elif isinstance(request, Request):
            django_request = _PhonyHttpRequest()
            django_request.META = request.headers
            if request.headers.get("X_DJANGO_OAUTH_TOOLKIT_SECURE", False):
                django_request._scheme = "https"
        else:
            raise TypeError("request must be a django or oauthlib request: got %r" % request)
        abs_url = django_request.build_absolute_uri(reverse("oauth2_provider:oidc-connect-discovery-info"))
        return abs_url[: -len("/.well-known/openid-configuration/")]
```